Add ASP.NET Core release notes for .NET 10 Preview 1 (#9740)
dotnet/core: release-notes/10.0/preview/preview1/aspnetcore.md

# ASP.NET Core updates in .NET 10 Preview 1

Here's a summary of what's new in ASP.NET Core in this preview release:

- [OpenAPI 3.1 support](#openapi-31-support)
- [Generate OpenAPI documents in YAML format](#generate-openapi-documents-in-yaml-format)
- [Response description on `ProducesResponseType`](#response-description-on-producesresponsetype)
- [Detect if URL is local using `RedirectHttpResult.IsLocalUrl`](#detect-if-url-is-local-using-redirecthttpresultislocalurl)

> - [Better support for testing apps with top-level statements](#better-support-for-testing-apps-with-top-level-statements)

Author: ```suggestion
- [Improvements to integration testing of apps with top-level statements](#improvements-to-integration-testing-of-apps-with-top-level-statements)
```

- [QuickGrid `RowClass` parameter](#quickgrid-rowclass-parameter)
- [Blazor script as a static web asset](#blazor-script-as-a-static-web-asset)
- [Route syntax highlighting for Blazor `RouteAttribute`](#route-syntax-highlighting-for-blazor-routeattribute)

ASP.NET Core updates in .NET 10 Preview 1:

- [Release notes](aspnetcore.md)
- [What's new in ASP.NET Core in .NET 10](https://learn.microsoft.com/aspnet/core/release-notes/aspnetcore-10.0) documentation.
- [Breaking changes](https://docs.microsoft.com/dotnet/core/compatibility/10.0#aspnet-core)

> - [Roadmap](https://aka.ms/aspnet/roadmap)

Author: ```suggestion
- [Roadmap](https://github.com/dotnet/aspnetcore/issues/59443)
```

For example:

```csharp
[HttpGet(Name = "GetWeatherForecast")]
[ProducesResponseType<IEnumerable<WeatherForecast>>(StatusCodes.Status200OK, Description = "The weather forecast for the next 5 days.")]
public IEnumerable<WeatherForecast> Get()
{
    ...
}
```

Generated OpenAPI response:

```json
"responses": {
  "200": {
    "description": "The weather forecast for the next 5 days.",
    "content": { ... }
  }
}
```

Thank you [sander1095](https://github.com/sander1095) for this contribution!

## Detect if URL is local using `RedirectHttpResult.IsLocalUrl`

Use the new `RedirectHttpResult.IsLocalUrl(url)` helper method to detect if a URL is local. A URL is considered local if it does not have the host or authority part and it has an absolute path. This method is useful for validating URLs before redirecting to them to prevent open redirection attacks.

```csharp
if (RedirectHttpResult.IsLocalUrl(url))
{
    return Results.LocalRedirect(url);
}
```

Thank you [@martincostello](https://github.com/martincostello) for this contribution!

> ## Better support for testing apps with top-level statements

Author: ```suggestion
## Improvements to integrating testing apps with top-level statements
```
